experimental/GModule/Cohomology.jl

```diff
@@ -763,11 +822,17 @@ function fp_group(M::GrpAbFinGen)
   return F, MapFromFunc(x->reduce(+, [sign(w)*gen(M, abs(w)) for w = word(x)], init = zero(M)), y->mF(reduce(*, [gen(G, i)^y[i] for i=1:ngens(M)], init = one(G))), F, M)
 end
 
+function underlying_word(g::FPGroupElem)
+  return FPGroupElem(free_group(parent(g)), GAP.Globals.UnderlyingElement(g.X))
+end
+
 """
 Given a 2-cocycle, return the corresponding group extension, ie. the large
-group, the injection of the abelian group and the quotient.
+group, the injection of the abelian group and the quotient as well as a map
+that given a tuple of elements in the group and the abelian group returns
+the corresponding elt in the extension. 
 """
-function extension(c::CoChain{2,PermGroupElem})
+function extension(c::CoChain{2,<:Oscar.GAPGroupElem})
```

Comment:
TODO: we need a version for this that takes PcGroups and produces PcGroups

Reply:
done